sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -540,22 +546,39 @@ public virtual ServiceBusReceivedMessage AmqpMessageToSBMessage(AmqpMessage amqp
                 }
             }
 
+            return annotatedMessage;
+        }
+
+        public virtual ServiceBusReceivedMessage AmqpMessageToSBReceivedMessage(AmqpMessage amqpMessage, bool isPeeked = false)
+        {
+            AmqpAnnotatedMessage annotatedMessage = AmqpMessageToAnnotatedMessage(amqpMessage, isPeeked);
+
+            ServiceBusReceivedMessage sbMessage = new ServiceBusReceivedMessage(annotatedMessage);
+
             // lock token
 
-            if (amqpMessage.DeliveryTag.Count == GuidSizeInBytes)
+            sbMessage.LockTokenGuid = ParseGuidBytes(amqpMessage.DeliveryTag);
+
+            amqpMessage.Dispose();
+
+            return sbMessage;
+        }
+
+        public virtual Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
+        {
+            if (bytes.Length == GuidSizeInBytes)
             {
                 Span<byte> guidBytes = stackalloc byte[GuidSizeInBytes];
-                amqpMessage.DeliveryTag.AsSpan().CopyTo(guidBytes);
+                bytes.Span.CopyTo(guidBytes);
                 if (!MemoryMarshal.TryRead<Guid>(guidBytes, out var lockTokenGuid))
```

Comment:
Why is this reading a copy and not the original bytes? i.e. the bytes.Span

Reply:
Good question - this code was not actually changed other than doing a slight refactor. I will see if this can be simplified.

Reply:
I think we do the copy because the AMQP message will be disposed before returning the SBReceivedMessage.

Reply:
That is my understanding; we need the copy so that we can dispose the AMQP message.

Reply:
The stack allocated span is destroyed even sooner (when this method returns), and so copying to the span cannot possibly help. But, I don't think you actually need any help. TryRead copies the bytes into the Guid struct. It does not hold to the  bytes passed into TryRead. Same with the Guid ctor below.

Reply:
Ah, I thought we were talking about the body, but I'm realizing now this is also about the GUID.   Based on what you pointed out with the [GUID constructor](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Guid.cs,b622ef5f6b76c10a), we can probably eliminate this entire block and just pass `bytes.Span` to the constructor.